### PR TITLE
Change requestjob_checks_warning to use simpler db_introspect API 

### PR DESCRIPTION
**Pull request by luca-drf**
This is how I'd use the simpler API proposed here: https://github.com/Ensembl/ensembl-prodinf-core/pull/15

### CODE REVIEW

File: src/ensembl/production/dbcopy/views.py [hunk 1]
```diff
@@ -16,14 +16,15 @@
 from django.core.exceptions import ValidationError
 import logging
 
-from django.contrib.auth.decorators import login_required
+# from django.contrib.auth.decorators import login_required
```
Comment (marcoooo) [inline src/ensembl/production/dbcopy/views.py:19]: Shouldn't let comments hanging around, if possible.

File: src/ensembl/production/dbcopy/views.py [hunk 2]
```diff
@@ -61,104 +60,57 @@ def requestjob_checks_warning(request):
...
         #   Diff with src_dbames
-        tgt_name_filter, target_name_match = get_filter_match(request.POST.getlist('tgt_db_name', []))
-        logger.debug("src_db_set %s", src_db_set)
-        logger.debug("target_db_set_match %s", target_name_match)
-        tgt_db_set_match = src_db_set
-        # TODO manage tgt_name_filter
-        if target_name_match:
-            tgt_db_set_match = set(target_name_match)
-            logger.debug("Updated src_db_set with renamed targets %s", target_name_match)
+        tgt_db_names = set(request.POST.getlist('tgt_db_name', [])) or src_db_set
```
Comment (marcoooo) [inline src/ensembl/production/dbcopy/views.py:85]: Python question, if default is `[]` then is the `or` called at all?

**Review by marcoooo — COMMENTED** [review]
Apart for two small comments seems all good and simpler! Always good for maintenance. Thanks!

File: src/ensembl/production/dbcopy/views.py (hunk 2 above)
Comment (luca-drf) [reply src/ensembl/production/dbcopy/views.py:85]: Is called when `request.POST.getlist('tgt_db_name', [])` returns an empty list. From Python docs:

> In the context of Boolean operations, and also when expressions are used by control flow statements, the following values are interpreted as false: `False`, `None`, numeric zero of all types, and empty strings and containers (including strings, tuples, lists, dictionaries, sets and frozensets). All other values are interpreted as true. User-defined objects can customize their truth value by providing a `__bool__()` method.

https://docs.python.org/3/reference/expressions.html#booleans

**Comment by luca-drf** [comment]
Requires https://github.com/Ensembl/ensembl-prodinf-core/pull/20

File: requirements.txt [hunk 3]
```diff
@@ -5,4 +5,4 @@ drf-nested-routers~=0.93.3
 drf-yasg~=1.20.0
 django-autocomplete-light~=3.8.2
 ensembl-prodinf-djcore@git+https://github.com/Ensembl/ensembl-prodinf-djcore.git@1.1#egg=ensembl-prodinf-djcore
-ensembl-prodinf-core@git+https://github.com/Ensembl/ensembl-prodinf-core.git@1.1#egg=ensembl-prodinf-core
+ensembl-prodinf-core@git+https://github.com/Ensembl/ensembl-prodinf-core.git@1.3#egg=ensembl-prodinf-core
```
Comment (marcoooo) [inline requirements.txt:8]: I presume the dependency could be change to 
ensembl-prodinf-core~=2.0.0 ?
Assuming this is the corresponding version of https://github.com/Ensembl/ensembl-prodinf-core/pull/20 I just validated.

**Review by marcoooo — CHANGES_REQUESTED** [review]
The dependency should change, we'll have to be careful and change other apps deps as well,

File: requirements.txt (hunk 3 above)
Comment (luca-drf) [reply requirements.txt:8]: Correct. I'll change it as soon as we have core libs on PyPI.

**Review by marcoooo — APPROVED** [review]
I assume you tested locally right? :-)

**Comment by luca-drf** [comment]
> I assume you tested locally right? :-) There are already some automated tests regarding this, though. Did you think of adding a few new ones?

Yes. I didn't add tests to this one. I think `requestjob_checks_warning()` should be refactored into smaller functions and tested separatedly.

**Pull request merged**